jsonrpc/ipc_all.go

```diff
@@ -0,0 +1,62 @@
+//go:build !windows
+// +build !windows
+
+package jsonrpc
+
+import (
+	"context"
+	"errors"
+	"net"
+	"os"
+	"path/filepath"
+)
+
+const (
+	// http://man7.org/linux/man-pages/man7/unix.7.html
+	maxUnixPathSize = int(108)
+)
+
+var errPathTooLong = errors.New("path too long")
+
+// createListener creates a Unix domain socket at the given path and sets proper permissions.
+func createListener(endpoint string) (net.Listener, error) {
+	var (
+		l   net.Listener
+		err error
+	)
+	// path + terminator
+	if len(endpoint)+1 > maxUnixPathSize {
+		return nil, errPathTooLong
+	}
+	err = preparePath(endpoint)
+	if err != nil {
+		return nil, err
+	}
+	l, err = net.Listen("unix", endpoint)
+	if err != nil {
+		return nil, err
+	}
+	// Set permissions for the socket file to read and write for the owner only (0o600)
+	err = os.Chmod(endpoint, 0o600) //nolint:gomnd
+	return l, err
+}
+
+func UnixDial(ctx context.Context, endpoint string) (net.Conn, error) {
```

Comment:
This is only used in tests. Can we move it to the `jsonrpc_test` package?

Reply:
It can be moved but it will also need to be compiled differently for windows, `ipc_test_windows.go` and `ipc_test_all.go` is fine?

Reply:
Hmm, I see `IpcDial` in `ipc_windows.go`. Assuming we don't change anything, does that need to be renamed to `UnixDial` for the tests to build on windows?

Reply:
yeah, my bad, forgot about that one when doing a naming refactor from unix to ipc .

Reply:
No worries, just wanted to make sure I was understanding. To be honest, we won't be able to guarantee any level of support for the windows code, so we might as well drop it. That should simplify things. Wdyt?

Reply:
Lets drop it then